### ISSUE TITLE
Primary Constructor with Self-Type Parameter Creates Uninstantiable Class Without Compiler Error

### ISSUE THREAD

**Issue by kado-kliewer** [dotnet/roslyn#75742]
When using primary constructors, it's possible to create a class that requires an instance of itself to be instantiated, creating an impossible initialization scenario that should be flagged as a compiler error.

Example:
```csharp
public class JobType(JobType type)  // Compiles but can never be instantiated
{
    public int Id { get; set; } = type.Id;
}

// Usage is impossible:
var job = new JobType(???); // Requires existing JobType instance

```

This creates a logical paradox - the class can never be instantiated because it always requires a pre-existing instance of itself to construct. This is similar to other scenarios where a type is made uninstantiable (like a class with only private constructors and no static factory methods), which do result in compiler errors.

Expected behavior:

The compiler should emit an error indicating that the class cannot be instantiated due to a circular dependency in its primary constructor.

Current behavior:
The code compiles successfully despite being impossible to use.

Suggested error message:

"Error CSxxxx: Type 'JobType' cannot be instantiated because its primary constructor requires an instance of itself"

**Comment by CyrusNajmabadi** [comment]
It's entirely legal to pass 'null' here, and many recursive structures might do just that.  note: that does mean the code within the type needs to be resilient to that initial state.

**Comment by kado-kliewer** [comment]
You're absolutely right - I overlooked that null is a valid value for reference types. So this would be perfectly valid:

``` csharp
var job = new JobType(null);
```

I withdraw my suggestion about this being a compiler error. Thanks for helping me understand this better!
